passionscotch/app/controllers/index.rb

```diff
@@ -0,0 +1,112 @@
+enable :sessions
+
+get '/' do
+  redirect 'login'
+end
+
+get '/login' do
+  erb :login
+end
+
+
+post '/login' do
+  user = User.find_by(email: params[:email])
+  if user.authenticate(params[:password])
+    session[:user_id] = user.id
+    redirect "/#{user.username}"
+  else
+    @login_error = "Sorry, that didn't work. Please try again."
+    erb :login
+  end
+end
+
+post '/register' do
+  new_user = User.create(username: params[:username], email: params[:email], password: params[:password], name: params[:name])
+
+
+  redirect '/login'
+end
+
+get '/reviews' do
+  @user = User.find(session[:user_id])
+  @scotch = Scotch.all
+  erb :reviews
+end
+
+
+
+
+get '/logout' do
+  logout_user = session[:user_id]
+  session[:user_id] = nil
+  redirect '/login'
+end
+
+post '/:id/add_review' do
+  @user = User.find(session[:user_id])
+  @scotch = Scotch.find(params[:id])
+  session[:scotch_id] = @scotch.id
```

Comment:
Why are you storing `session[:scotch_id] = @scotch.id` in the session? This is a code smell. Find me if you want to chat about it.